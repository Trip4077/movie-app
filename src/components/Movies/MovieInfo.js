import React from 'react';

const MovieInfo = props => {
    return(
        <div className='movie-info'>
            <img src={props.movie.Poster} alt={props.movie.Title} />

            <table className='striped info-table'>
                <tbody>
                    <tr>
                        <td>Title:</td>
                        <td>{props.movie.Title}</td>
                    </tr>
                    <tr>
                        <td>Synopsis:</td>
                        <td>{props.movie.Plot}</td>
                    </tr>
                    <tr>
                        <td>Rating:</td>
                        <td>{props.movie.Rated}</td>
                    </tr>
                    <tr>
                        <td>Genre:</td>
                        <td>{props.movie.Genre}</td>
                    </tr>
                    <tr>
                        <td>Runtime:</td>
                        <td>{props.movie.Runtime}</td>
                    </tr>
                    <tr>
                        <td>Release Date:</td>
                        <td>{props.movie.Released}</td>
                    </tr>
                    <tr>
                        <td>Cast:</td>
                        <td>{props.movie.Actors}</td>
                    </tr>
                    <tr>
                        <td>Production:</td>
                        <td>{props.movie.Production}</td>
                    </tr>
                    <tr>
                        <td>Box Office:</td>
                        <td>{props.movie.BoxOffice}</td>
                    </tr>
                </tbody>
            </table>
        </div>
    )
}

export default MovieInfo;