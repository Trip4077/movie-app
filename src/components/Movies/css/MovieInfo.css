/*
---------------------------MovieInfo Styles---------------------------
*/

.movie-page {
    margin-bottom: 60px;
  }
  
  .movie-info {
    display: flex;
    justify-content: space-between;
  
    margin-top: 50px;
  }
  
  .info-table {
    margin-left: 25px;
  }
  
  .info-table td:first-of-type {
    font-weight: bold;
  }
  
  .scheduler, .picker-container {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
  
    width: 100%;
  }
  
  .xxx, .react-datepicker__input-container {
    width: 70%;
    display: flex;
    justify-content: center;
  }
  
  .picker-container .react-datepicker-wrapper {
    display: flex;
    justify-content: center;
    width: 100%;
  }
  
  .picker-container h6 {
    align-self: flex-start;
  
    margin: 40px 0 0 14.5%;
    font-weight: bold;
  }
  
  .movie-page button {
    margin-top: 30px;
  }