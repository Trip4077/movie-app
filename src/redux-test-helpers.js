import React from 'react';

import { createStore } from 'redux';
import { Provider } from 'react-redux';

import { MemoryRouter } from 'react-router';

import { render } from 'react-testing-library';

import * as ACTIONS from './actions';

function reducer(state = { userReducer: { user: { id: 1 }}}, action) {
    switch(action.type) {
        case ACTIONS.LOADING:
            return {
                ...state,
                loading: true
            }

        case ACTIONS.END_LOAD:
            return {
                ...state,
                loading: false
            }

        case ACTIONS.UPDATE_FAV_LIST:
            return {
                loading: false,
                favorites: action.payload,
                schedule: state.schedule
            }

        case ACTIONS.GET_INFO:
            return {
                ...state,
                loading: false,
                info: action.payload
            }

        case ACTIONS.UPDATE_SCHEDULE:
            return {
                loading: false,
                favorites: state.favorites,
                schedule: action.payload
            }

        case ACTIONS.TEXT_SENT:
            return {
                ...state,
                loading: false
            }

            case ACTIONS.REGISTER: 
            return {
                loading: false,
                userID: action.payload
            }

        case ACTIONS.LOGIN:
            localStorage.setItem( 'token', action.payload.token )

            delete action.payload.user.password

            return {
                loading: false,
                user: action.payload.user
            }

        case ACTIONS.LOGOUT:
            return { ...initialState }

        default:
            return {
                ...state
            }
    }
}

export function renderWithRedux( 
    ui, 
    { initialState, store = createStore(reducer, initialState) } = {} ) {
        return {
            ...render( <Provider store={store}>
                         <MemoryRouter>
                            <div> {ui} </div>
                         </MemoryRouter>
                       </Provider> ),
            store
        }
    };