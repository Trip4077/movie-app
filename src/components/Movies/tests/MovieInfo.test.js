import React from 'react';

import {  cleanup } from 'react-testing-library';
import { renderWithRedux } from '../../../redux-test-helpers';

import pear from 'react-test-renderer';

import MovieInfo from '../MovieInfo';

afterEach(cleanup);

describe('<MovieInfo />', () => {
  const movie = {
     Poster: 'link',
     Title: 'Avengers',
     Plot: 'Fight Bad Guys',
     Rated: 'PG13',
     Genre: 'Action',
     Runtime: 'Couple Hours',
     Released: '12-12-1212',
     Actors: 'RDJ',
     Production: 'Disney',
     BoxOffice: 'a lot'
  }

  it('renders without crashing', () => {
    renderWithRedux(<MovieInfo movie={movie} />);
  });

  it('renders a "Title" row', () => {
      const { getByText } = renderWithRedux(<MovieInfo movie={movie} />);

      getByText(/title:/i);
  });

  it('renders a "Synopsis" row', () => {
    const { getByText } = renderWithRedux(<MovieInfo movie={movie} />);

    getByText(/synopsis:/i);
  });

  it('renders a "Rating" row', () => {
    const { getByText } = renderWithRedux(<MovieInfo movie={movie} />);

    getByText(/rating:/i);
  });

  it('renders a "Genre" row', () => {
    const { getByText } = renderWithRedux(<MovieInfo movie={movie} />);

    getByText(/genre:/i);
  });

  it('renders a "Runtime" row', () => {
    const { getByText } = renderWithRedux(<MovieInfo movie={movie} />);

    getByText(/runtime:/i);
  });

  it('renders a "Release  Date" row', () => {
    const { getByText } = renderWithRedux(<MovieInfo movie={movie} />);

    getByText(/release date:/i);
  });

  it('renders a "Cast" row', () => {
    const { getByText } = renderWithRedux(<MovieInfo movie={movie} />);

    getByText(/cast:/i);
  });

  it('renders a "Production" row', () => {
    const { getByText } = renderWithRedux(<MovieInfo movie={movie} />);

    getByText(/production:/i);
  });

  it('renders a "Box Office" row', () => {
    const { getByText } = renderWithRedux(<MovieInfo movie={movie} />);

    getByText(/box office:/i);
  });

  it('should match snapshot - with props only', () => {
    const tree = pear.create(<MovieInfo movie={movie} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});