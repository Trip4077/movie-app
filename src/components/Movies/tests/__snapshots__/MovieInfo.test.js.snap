// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieInfo /> should match snapshot - with props only 1`] = `
<div
  className="movie-info"
>
  <img
    alt="Avengers"
    src="link"
  />
  <table
    className="striped info-table"
  >
    <tbody>
      <tr>
        <td>
          Title:
        </td>
        <td>
          Avengers
        </td>
      </tr>
      <tr>
        <td>
          Synopsis:
        </td>
        <td>
          Fight Bad Guys
        </td>
      </tr>
      <tr>
        <td>
          Rating:
        </td>
        <td>
          PG13
        </td>
      </tr>
      <tr>
        <td>
          Genre:
        </td>
        <td>
          Action
        </td>
      </tr>
      <tr>
        <td>
          Runtime:
        </td>
        <td>
          Couple Hours
        </td>
      </tr>
      <tr>
        <td>
          Release Date:
        </td>
        <td>
          12-12-1212
        </td>
      </tr>
      <tr>
        <td>
          Cast:
        </td>
        <td>
          RDJ
        </td>
      </tr>
      <tr>
        <td>
          Production:
        </td>
        <td>
          Disney
        </td>
      </tr>
      <tr>
        <td>
          Box Office:
        </td>
        <td>
          a lot
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
